File: remove_duplicates_from_sorted_array.py
class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        if len(A) <= 1: return len(A)
        i = 0
        for j in xrange(1, len(A)):
            if A[i] != A[j]:
                i += 1
                A[i] = A[j]

        return i + 1

File: various/ruby/easy/remove_duplicates_from_sorted_array.rb
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/description/
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
  return nums if nums.size <= 1
  j = 0
  for i in 1..nums.size-1
    if nums[i] != nums[j]
      j += 1
      nums[j] = nums[i]
    end
  end

  nums
end

p remove_duplicates([1,1,1,2,3,3,4,4,5])
